### DISCUSSION TITLE
How to Eager Load more than 1 relationship

### DISCUSSION THREAD

**Discussion by yswtrue** [SeaQL/sea-orm#1114] [Ideas]
## Motivation

Like there are 3 classes, A, B, C
and A has a one-to-one with B and C.

How can I Eager Load B and C relations by A.

I saw `find_also_related` only supports one relationship.

**Comment by billy1624** [comment]
Hey @yswtrue, welcome! Please check https://www.sea-ql.org/blog/2022-05-14-faq-01/ for the explaination in details.

Related discussions:
- https://github.com/SeaQL/sea-orm/discussions/1044
- https://github.com/SeaQL/sea-orm/discussions/649

**Comment by tyt2y3** [comment]
I think basically we want a [DataLoader](https://docs.rs/async-graphql/latest/async_graphql/dataloader/struct.DataLoader.html) to achieve Eloquent-like [eager loading](https://laravel.com/docs/9.x/eloquent-relationships#eager-loading).
I am only puzzling with how to define the shape of the output struct, because in Rust we can't just set an attribute dynamically. My current conclusion is that we need a proper GraphQL `Node` implementation. So it might be interesting to see if it's possible to bring in some bits and pieces from Seaography.

**Reply by phungleson** [reply]
I think lacking of a `dynamic` data modelling makes `sea-orm` a bit hard to use in real world. As in real world, we would normally do custom select + eager loading. Especially in area like GraphQL where you can specify what to return.

DataLoader like one in `async-graphql` is tricky as it is designed to be general with potential long delay. I am not sure if it fits for the use of SQL, which I guess, need to be more deterministic?

Not sure if it is easier to think about the problem by breaking it into steps? instead of trying to solve everything in a giant query.

DSL + query + data mapping

**Reply by tyt2y3** [reply]
If you are building GraphQL resolvers definitely checkout Seaography

**Reply by phungleson** [reply]
Thanks I checked out Seaography.

From 10k above it seems that it does more than what I need. For example my authorization is complex, i.e. certain people can do certain thing to certain models. Seaography seems to gen all the GraphQL code, I am not sure how to put my authorization layer in. Is there an example to use the entities without the resolvers?

I guess I need a complex SQL query engine but without eager loading, I am not sure if it worth to use `sea-orm` just for the DSL, since I am already use `sea-query`? i.e. performance wise, it is probably worst since it is more abstraction?

**Reply by frederikhors** [reply]
@phungleson I'm in your situation. I just opened https://github.com/SeaQL/sea-orm/discussions/1139 and found out this discussion.

Have you found a solution?

**Reply by frederikhors** [reply]
@phungleson sorry to bother you. Did you find a solution?

**Reply by phungleson** [reply]
Hey @frederikhors no I couldn't use much of `sea-orm`.

I am doing all queries in `sea-query` and eager load whenever necessary.
